packages/client/ui/react-ui/src/components/auth/PasskeyPrompt.tsx

```diff
@@ -0,0 +1,266 @@
+        >
+            {/* This is a hacky workaround to have the foreground transparent styles to work. */}
+            <Transition.Child
+                as={Fragment}
+                enter="ease-out duration-400"
+                enterFrom="opacity-0"
+                enterTo="opacity-100"
+                leave="ease-in duration-400"
+                leaveFrom="opacity-100"
+                leaveTo="opacity-0"
+            >
+                <div
+                    className={classNames(
+                        "fixed inset-0 bg-[rgba(139,151,151,0.2)] backdrop-blur-[2px] transition-opacity duration-300 ease-in-out -z-10"
+                    )}
+                />
+            </Transition.Child>
+            <div
+                className={classNames(
+                    "flex flex-col items-center w-full max-w-[28rem] rounded-2xl shadow-sm z-30 border",
+                    appearance?.colors?.background != null ? `bg-[${appearance.colors.background}]` : "bg-white",
+                    appearance?.colors?.border != null ? `border-[${appearance.colors.border}]` : "border-[#D0D5DD]"
+                )}
+                onClick={(e) => e.stopPropagation()}
+            >
+                <div className="pt-12 pb-10 px-8">
+                    <div className="flex justify-center left-1.5 relative">
+                        <PasskeyPromptLogo appearance={appearance} />
+                    </div>
+
+                    <div className="flex justify-center">
+                        <p
+                            className={classNames(
+                                "text-lg font-bold",
+
+                                appearance?.colors?.textPrimary != null
+                                    ? `text-[${appearance.colors.textPrimary}]`
+                                    : "text-[#20343E]"
+                            )}
+                        >
+                            {title}
+                        </p>
+                    </div>
+
+                    <div className="mt-4 mb-9">
+                        <div
+                            className={classNames(
+                                "font-normal",
+                                appearance?.colors?.textSecondary != null
+                                    ? `text-[${appearance.colors.textSecondary}]`
+                                    : "text-[#67797F]"
+                            )}
+                        >
+                            {content}
+                        </div>
+                    </div>
+
+                    <div className="flex flex-col gap-4 justify-center">
+                        {primaryButton}
+                        {secondaryAction}
+                    </div>
+                    <div className="flex justify-center pt-4">
+                        <PoweredByCrossmint color={appearance?.colors?.textSecondary} />
+                    </div>
+                </div>
+            </div>
+        </Dialog>
+    );
+}
```

Comment:
this component is big and really hard to read because there's a lot of customization at each step.
Could we consider breaking it up into smaller sub-components?

```
type Props = ...
function PasskeyPromptCore({ title, content, primaryButton, secondaryAction, appearance }: Props) {
    return (
        <Dialog
            open
            onClose={() => {}}
            className={classNames("fixed inset-0 z-20 flex items-center justify-center overflow-y-auto")}
        >
            {/* This is a hacky workaround to have the foreground transparent styles to work. */}
            <HackTransparentForegroundWrapper />
            <ModalWrapper>
                <div className="pt-12 pb-10 px-8">
                    <div className="flex justify-center left-1.5 relative">
                        <PasskeyPromptLogo appearance={appearance} />
                    </div>

                    <PromptTitle title={title}/>
                    <PromptContent> / <PromptContentWrapper content={content}>

                    <div className="flex flex-col gap-4 justify-center">
                        {primaryButton}
                        {secondaryAction}
                    </div>
                    <div className="flex justify-center pt-4">
                        <PoweredByCrossmint color={appearance?.colors?.textSecondary} />
                    </div>
                </div>
            </div>
        </Dialog>
    );
}
```

Reply:
Good callout. While I generally favor abstraction and smaller components, I believe this prompt component is an exception where breaking it down won't necessarily improve readability, especially for future maintenance and updates. Multiple sub-components can actually worsen performance due to prop drilling and potentially bloat the file. Following AHA (Avoid Hasty Abstractions) principles often leads to better React components in the long run. Plus, using Tailwind already helps us save a ton of lines, reducing the need for further componentization.

Reply:
There were a bunch of empty lines that I removed which makes this even smaller 😎